### PR TITLE
fix: check if typeof Auth0Widget is undefined

### PR DESCRIPTION
**Pull request by Zertz**
This partially fixes issue #129

### CODE REVIEW

**Comment by Zertz** [comment]
In order to "fix" the issue where `Auth0Lock` is not found, I have replaced `module.exports = Auth0Lock;` by `window.Auth0Lock = Auth0Lock;`

I'm sure this breaks other things so the Auth0 folks really need to implement the actual fix for that! (Using bower is only a workaround, not any better than my hacky solution IMO)

**Comment by mgonto** [comment]
Thanks for sending this!

@cristiandouce Thoughts?

**Comment by cristiandouce** [comment]
@Zertz what do you mean by partially?

I read this from #129 

```
if (null != Auth0Widget) {
  throw new Error('Auth0Widget is not supported with this version of auth0-angular' + 'anymore. Please try with an older one');
}
the Auto0widget is undefined
```

But the truth table for null is the following:

```js
null == undefined // true
null == null // true
null != undefined // false
null != null // false
```

Therefore, if Auth0Lock is actually undefined, that line should never trigger.

I belive the issue here is that Auth0Lock __is__ `undefined` and not that null check there.

Could you tell how this PR is a fix of that issue?

also, when we build our bundle we use this file as entrypoint:

https://github.com/auth0/lock/blob/621333319910bdb7ea8a744002e4e99f1d7b8c64/standalone.js

which, as you might see does expose to the `global` object the Auth0Lock function, which then ends up in the window object by hand of our browserify bundler.

The problem I think could be that `require` or either `define` do exist in the window/global objects at the moment this is run, and then avoids to define it globally-

What I mean is this:

```html
<script src="/my/bundle/app-with-require.js"></script>
<script src="https://cdn.auth0.com/js/lock-7.js"></script>
```

This way the lock may wrongly attempt to define itself in the require/define function from your app.

I'm not sure this is the actual issue, but either I see how this check change could fix it.

Please let me know your thoughts!

**Comment by Zertz** [comment]
Just to be clear, the PR fixes `Auth0Widget is not defined` and properly throws `Error('Auth0Widget is not supported with this version of auth0-angular anymore. Please try with an older one')` or `Error('Cannot initialize Auth0Angular. Auth0Lock or Auth0 must be available')` instead.

I realize that `null == undefined`, but if you try to run an Angular app **without** including `auth0-lock` you'll see that JavaScript chokes on `null != Auth0Widget`. Checking for `typeof Auth0Widget !== 'undefined'` fixes that issue.

**Comment by nolazybits** [comment]
just updated my bower and had to look in my browser history to find the fix again... I wished this pr would be accepted already

File: src/auth0-angular.js [hunk 1]
```diff
@@ -179,12 +179,12 @@
         };
       }
 
-      if (null != Auth0Widget) {
+      if (typeof Auth0Widget !== 'undefined') {
```
Comment (cristiandouce) [inline src/auth0-angular.js]: how does this resolve different?
It's the same truth table AFAIK

Comment (Zertz) [inline src/auth0-angular.js]: I can't give you the technical explanation, but no matter what the truth table says, the code in its current state is broken and that PR fixes it. Just try it.

Comment (charlierudolph) [inline src/auth0-angular.js]: Running in the browser console
```
> null != Auth0Widget
Uncaught ReferenceError: Auth0Widget is not defined(…)(
> typeof Auth0Widget !== 'undefined'
false
```

Comment (Zertz) [inline src/auth0-angular.js]: @cristiandouce Keep in mind that `null !== undefined` and `Auth0Widget` has never been defined in that context and thus it cannot simply be compared with `null`. Additionally, you should *always* use strict equality in JavaScript.

Up to you guys of course, but for the sake of your users this change should be merged!

**Comment by TrevorBasinger** [comment]
I'd also like for this to be merged. Is there a reason why it's not yet?

**Comment by lobo** [comment]
I've been asked by a client to make this merged. Any reason for it not to be merged? Thanks in advance.

**Comment by daviddk** [comment]
Same issue here, a merge would be nice indeed.

**Pull request merged**